pallets/gear/src/benchmarking/code.rs

```diff
@@ -0,0 +1,508 @@
+{
+    fn from(def: ModuleDefinition) -> Self {
+        // internal functions start at that offset.
+        let func_offset = u32::try_from(def.imported_functions.len()).unwrap();
+
+        // Every contract must export "init" and "handle" functions
+        let mut contract = builder::module()
+            // init function (first internal function)
+            .function()
+            .signature()
+            .build()
+            .with_body(
+                def.init_body
+                    .unwrap_or_else(|| FuncBody::new(Vec::new(), Instructions::empty())),
+            )
+            .build()
+            // handle function (second internal function)
+            .function()
+            .signature()
+            .build()
+            .with_body(
+                def.handle_body
+                    .unwrap_or_else(|| FuncBody::new(Vec::new(), Instructions::empty())),
+            )
+            .build()
+            .export()
+            .field("init")
+            .internal()
+            .func(func_offset)
+            .build()
+            .export()
+            .field("handle")
+            .internal()
+            .func(func_offset + 1)
+            .build();
+
+        // If specified we add an additional internal function
+        if let Some(body) = def.aux_body {
+            let mut signature = contract.function().signature();
+            for _ in 0..def.aux_arg_num {
+                signature = signature.with_param(ValueType::I64);
+            }
+            contract = signature.build().with_body(body).build();
+        }
+
+        // Grant access to linear memory.
+        if let Some(memory) = &def.memory {
+            contract = contract
+                .import()
+                .module("env")
+                .field("memory")
+                .external()
+                .memory(memory.min_pages, Some(memory.max_pages))
+                .build();
+        }
+
+        // Import supervisor functions. They start with idx 0.
+        for func in def.imported_functions {
+            let sig = builder::signature()
+                .with_params(func.params)
+                .with_results(func.return_type.into_iter().collect())
+                .build_sig();
+            let sig = contract.push_signature(sig);
+            contract = contract
+                .import()
+                .module(func.module)
+                .field(func.name)
+                .with_external(elements::External::Function(sig))
+                .build();
+        }
+
+        // Initialize memory
+        for data in def.data_segments {
+            contract = contract
+                .data()
+                .offset(Instruction::I32Const(data.offset as i32))
+                .value(data.value)
+                .build()
+        }
+
+        // Add global variables
+        if def.num_globals > 0 {
+            use rand::{distributions::Standard, prelude::*};
+            let rng = rand_pcg::Pcg32::seed_from_u64(3112244599778833558);
```

Comment:
Why this very number? Maybe it's worth introduce a constant for it with description?

Reply:
This is just a random seed to make sure the globals don't change on every run.